pkg/core/lifecycle.go

```diff
@@ -236,6 +236,14 @@ func (pl *Pipeline) Start(ctx context.Context) (err error) {
 			return err
 		}
 
+		pl.Logger.Debugf("Caching workspace")
```

Comment:
why this change require ? persisting before discovery how does it help us ?

Reply:
Please read the attached issue to get more context.